fastlane/helpers/android_virtual_device_helper.rb

```diff
@@ -2,12 +2,45 @@ module Fastlane
   module Helpers
     module AndroidVirtualDevicePathHelper
       def self.android_home
-        ENV['ANDROID_HOME'] || ENV['ANDROID_SDK_ROOT'] || ENV['ANDROID_SDK']
+
+        android_sdk_root = ENV['ANDROID_HOME'] || ENV['ANDROID_SDK_ROOT'] || ENV['ANDROID_SDK']
+
+        if android_sdk_root
+            return android_sdk_root
+        end
+
+        begin
+            require 'java-properties'
+        rescue LoadError
+            UI.user_error! "Unable to load dependencies. Did you run `bundle` and `bundle exec fastlane`?"
+        end
+
+        propertiesFile = "#{project_root_path}/local.properties"
+
+        local = nil
+
+        if File.exist? propertiesFile
+            properties = JavaProperties.load(propertiesFile)
+            local = properties[:"sdk.dir"]
+        end
+
+        if android_sdk_root.nil? && local.nil?
+            UI.user_error! "There is no Android SDK root set. Unable to continue."
+        elsif android_sdk_root.nil? && local
+            UI.user_error! "There is no Android SDK root set. Unable to continue.\nTo solve this, add:\n\nexport ANDROID_HOME=#{local}\nexport ANDROID_SDK_ROOT=$ANDROID_HOME\n\nto ~/.profile"
+
```

Comment:
Can we remove this empty line?
(Sorry for this one, but at this point we started to fix this small layout issue... let's make it perfect :-) )

Reply:
Just to clarify - just the top line? Or all the empty lines?

Reply:
No, just the empty line in this `elsif` statement. The one before the `end`. Line 31 as GH marks it.

Reply:
Glad I asked 😅 

It's in https://github.com/wordpress-mobile/WordPress-Android/pull/8211/commits/1dd52398cc9d862d274e16bc72af8a67d071d7c4! 😄